### PR TITLE
Small changes and a bugfix

### PR DESCRIPTION
**Pull request by lord**
I'm not very familiar with Python, so let me know if any of this is weird!

The quotes are because there are commas in the address field, and they need to be escaped.

### CODE REVIEW

File: directory.py [hunk 1]
```diff
@@ -85,8 +85,9 @@ def get_people(d):
         floor = d[k]['floor']
         roomNumber = d[k]['roomNumber']
         email = d[k]['photo']
+        address = d[k]['address']
         urllib.urlretrieve("https://apps.carleton.edu/stock/ldapimage.php?id=%s&source=campus_directory" %email, "stalkernet_images/%s.jpg" %email)
-        output_file.write("%s,%s,%d,%s,%d,%s,%s\n" %(name, majors, year, dorm, floor, roomNumber, email))
+        output_file.write("%s,%s,%d,%s,%d,%s,%s,\"%s\"\n" %(name, majors, year, dorm, floor, roomNumber, email, address))
```
Comment (Machyne) [inline directory.py]: As long as you're touching this line, mind changing it to single quote and using `str.format`? Maybe something like
````
output_file.write('{0},{1},{2},{3},{4},{5},{6},"{7}",\n'.format(
    name, majors, year, dorm, floor, roomNumber, email, address))
`````

File: directory.py [hunk 2]
```diff
@@ -149,7 +155,8 @@ def data_of(x):
         'dorm': dorm,
         'floor': floor,
         'roomNumber': roomNumber,
-        'photo': photo
+        'photo': photo,
+        'address': address
```
Comment (Machyne) [inline directory.py]: Can you add a `,` at the end here (i.e. `'address': address,`)? That way if someone adds more fields (phone numbers??) there won't be a diff on this line.

File: directory.py [hunk 3]
```diff
@@ -105,7 +106,11 @@ def data_of(x):
     major = re.findall(re_major, x)
     concentration = re.findall(re_concentration, x)
     year = re.search(re_year, x)
-    year = int(year.groups()[0]) if year else None
```
Comment (Machyne) [inline directory.py]: What was the bug here, out of curiosity?

Comment (lord) [inline directory.py]: Somebody's year wasn't able to parse into a string, so there was a `ValueError`? Not sure if this is the best way to do it. I think the year had a bunch of crazy characters in it/was an edge case.

File: directory.py (hunk 2 above)
Comment (lord) [inline directory.py]: :+1:

**Comment by lord** [comment]
Ack! Didn't mean to merge! Accidentally pushed to the wrong branch, sorry. Would rather not force push `master`. Should I do a revert commit, or did the new changes fix the issues properly?

**Comment by Machyne** [comment]
I think if you changed [line 43](https://github.com/CarletonDevX/Stalkernet/blob/8baaf565529e043bcce3e965d18f9e6fef4acd2b/directory.py#L43) to
````
re_year = '<span class="affiliation">(\d*?)</span>'
````
it would fix the year parsing issue.

As for force push vs revert, I think @chetaldrich or @lbiester should decide.

**Comment by lord** [comment]
IIRC correctly, it looked something like `<span class="affiliation">(2016) / Alumni</span>` — just realizing now that perhaps this is someone who graduated early.

Maybe  `re_year = '<span class="affiliation">.*(\d*?).*</span>`?

**Comment by lord** [comment]
And @lbiester @chetaldrich — just let me know what I should do! Sorry for mucking up the repo. :cry:

**Comment by AdamCanady** [comment]
For that regex, maybe `re_year = '<span class="affiliation">[^<]*(\d*?)[^<]*</span>` or better yet `re_year = '<span class="affiliation">.*?(\d*?).*?</span>`. That `*` operator is greedy and will mess things up depending on how things are scoped. The `?` tells it to stop as soon as it can.

**Pull request merged**